wordpress-mobile/WordPress-Android
# Jetpack Section: Backup Screen - Introduce Backup Screen

Parent #13629 

This PR introduces the `Backups` screen, which is actually an `Activity Log` screen in disguise. As such, this PR reused the `Activity Log` screen to show the `Backups` screen. The below list of items change when launching an the `Activity Log` for the `Backups` screen:
* The screen title changes to `Backup`.
* The `Activity Type` chip is being hidden.
* The list of activities only contain `Rewindable` items.
* The empty screen state without filter changes to `Your first backup will be ready soon`.
* The empty screen state with filter changes to `No matching backups found`.

![screen-20210113-175331 (3)](https://user-images.githubusercontent.com/9729923/104480519-5523d380-55cd-11eb-9029-b0b669b4f1ed.gif)

To test:
* Launch app and go to `My Site` tab.
* Go to `My Site` -> `Toolbar Avatar` -> `App Settings` -> `Test feature configuration`.
* Enable `BackupsFeatureConfig`, scroll down and click the `RESTART THE APP` button. **DO NOT** confuse this feature flag with the `BackupFeatureConfig` (singular). Make sure to enable the right feature flag.
* Go back to `My Site` tab and notice `Backup` menu is shown.
* Select the `Activity` menu and verify UI correctness (see above) and that all features work as expected. This is includes but is not limited to:
  * Scrolling infinitely, from top to bottom and vice-versa.
  * Triggering the `Date Range` and `Activity Type` filters.
  * Selecting an activity and navigating to its detail page.
  * Triggering `Restore to this point` and `Download backup` from the menu item.
  * While restoring to a point observe the progress item.
  * Offline support.
  * Configuration changes (rotation, language, etc).
* Select the `Backup` menu and verify UI correctness (see above) and that all features work as expected. This is includes but is not limited to:
  * Scrolling infinitely, from top to bottom and vice-versa. ⚠️ - Scrolling is not optimized yet, try an break it... more on that later
  * Triggering the `Date Range` filter.
  * Selecting an activity and navigating to its detail page.
  * Triggering `Restore to this point` and `Download backup` from the menu item.
  * While restoring to a point observe the progress item.
  * Offline support.
  * Configuration changes (rotation, language, etc).

This solution is based on the investigation notes [here](https://github.com/wordpress-mobile/WordPress-Android/issues/13629#issuecomment-758531526). Below is a summary of the decisions that was made based on that:
* Filter the list of activities after storing them in db. The necessary API enhancement is not available yet and as such filtering it before would create unnecessary overhead and problems later on.
* Do not enhance the scrolling behavior since there will be anyway cases where `filtering after storing in db` approach chosen above would decrease user experience (ie. 1 rewindable item per 20 items per page) or even worse break it altogether (ie. 0 rewindable items per 20 items per page). As such, the API enhancement is unavoidable and it is already being discussed.
* Hide the `Activity Type` filter since the API for this filter is only available for all items. An intermediate solution is still viable, but it has been decided to just hide this filter for now.
* Reuse `Activity Log` screen since by not doing so this will create lots of duplicate code.

Future:
* Add tracking.
* Polish UI.
* Accessibility.
* Design review.

PR submission checklist:
